test/serverExplorer.test.ts

```diff
@@ -0,0 +1,140 @@
+        serverExplorer.insertServer(serverHandle);
+        expect(refreshStub).calledOnce;
+    });
+
+    test('should able to removeServer the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const disposeStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+            dispose: () => {}
+        });
+        sandbox.stub(serverExplorer, 'refresh')
+        serverExplorer.removeServer(serverHandle);
+        expect(disposeStub).calledOnce;
+    });
+
+    test('should able to updateServer the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const serviceStub = sinon.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+        });
+        sandbox.stub(serverExplorer, 'refresh')
+        serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to showOutput of the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to getTreeItem of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverStatus, 'get').returns({
+            serverHandle
+        });
+        serverExplorer.getTreeItem(serverHandle);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to getChildren of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.servers, 'values').returns([]);
+        const getChildren = serverExplorer.getChildren(undefined);
+        expect(getChildren).deep.equals(Array.from(clearStub));
+    });
+
+    test('should able to getTreeItem of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns(undefined);
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to refresh of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(EventEmitter.prototype, 'fire');
+        serverExplorer.refresh(serverHandle);
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(serverHandle);
+    });
+
+    test('should able to addLocation for the server', async () => {
```

Comment:
rename to: 'serverExplorer.addLocation should ask for location of the server and name if server detected in provided location`